src/cpp/src/tokenizer/tokenizer.cpp

```diff
@@ -471,16 +477,66 @@ class Tokenizer::TokenizerImpl {
         );
     }
 
-    TokenizedInputs encode(std::vector<std::string>& prompts, const ov::AnyMap& tokenization_params = {}) {
+    TokenizedInputs encode(const std::vector<std::pair<std::string, std::string>>& prompts_pairs, const ov::AnyMap& tokenization_params = {}) {
+        OPENVINO_ASSERT(m_ireq_queue_tokenizer, "Either openvino_tokenizer.xml was not provided or it was not loaded correctly. "
+                                                "Tokenizer::encode is not available");
+        std::vector<ov::Tensor> inputs(2);
+        size_t batch_size = prompts_pairs.size();
+        std::vector<std::string> prompts_1(batch_size);
+        std::vector<std::string> prompts_2(batch_size);
+
+        for(size_t i = 0; i < batch_size; ++i) {
+            prompts_1[i] = prompts_pairs[i].first;
+            prompts_2[i] = prompts_pairs[i].second;
+        }
+        return encode(prompts_1, prompts_2, tokenization_params);
+    }
+
+    TokenizedInputs encode(const std::vector<std::string>& prompts_1, const std::vector<std::string>& prompts_2, const ov::AnyMap& tokenization_params = {}) {
+        OPENVINO_ASSERT(m_ireq_queue_tokenizer, "Either openvino_tokenizer.xml was not provided or it was not loaded correctly. "
+                                                "Tokenizer::encode is not available");
+        std::vector<ov::Tensor> prompts(2);
+        size_t batch_size = prompts_1.size();
+        OPENVINO_ASSERT(prompts_1.size() == prompts_2.size() || prompts_1.size() == 1 | prompts_2.size() == 1, 
+                        "prompts_1 and prompts_2 should be of the same size or one of them should be of size 1");
+        prompts[0] = ov::Tensor{ov::element::string, {batch_size}, const_cast<std::string*>(prompts_1.data())};
+        prompts[1] = ov::Tensor{ov::element::string, {batch_size}, const_cast<std::string*>(prompts_2.data())};
+        std::vector<std::string> prompts_1_, prompts_2_;
+        return encode(prompts, tokenization_params);
+    }
+
+    TokenizedInputs encode(const std::vector<ov::Tensor>& prompts, const ov::AnyMap& tokenization_params = {}) {
```

Comment:
```suggestion
    TokenizedInputs encode(const std::array<ov::Tensor, 2>& prompts, const ov::AnyMap& tokenization_params = {}) {
```
Or `pair`, That way the assert isn't needed. The second Tensor can be default constructed if it's not required

Reply:
I removed this overload. Since it's used only in one place i moved code inside other function. Not passing between functions is needed now